docs/content/ecs/component-generation.md

```diff
@@ -13,27 +13,49 @@ The [code generator]({{urlRoot}}/content/code-generator) uses `.schema` files to
 
 ## Overview
 
-A `struct`, which implements `Unity.Entities.IComponentData` and `Improbable.Gdk.Core.ISpatialComponentData`, is generated for each SpatialOS component. It has the same name as the corresponding schema component.
-The generation process creates a namespace for each struct according to the relevant schemalang component name: `{name of schema component}.Component`
+The code generator generates two structs for each SpatialOS component defined in schema. The generation process places these structs in a namespace according to the SpatialOS component name: `{schema package}.{name of schema component}`. The two structs it generates are:
+
+  * A `struct` which implements `Improbable.Gdk.Core.ISpatialComponentSnapshot`. Its fully qualified name is: `{schema package}.{name of schema component}.Snapshot`.
+  * A `struct` which implements `Unity.Entities.IComponentData`, `Improbable.Gdk.Core.ISpatialComponentData`, and `Improbable.Gdk.Core.ISnapshottable<{schema package}.{name of schema component}.Snapshot>`. Its fully qualified name is: `{schema package}.{name of schema component}.Component`.
+
 These structs only contain the defined schema data fields. They do *not* contain any fields or methods relating to [commands]({{urlRoot}}/content/ecs/sending-receiving-component-commands) or [events (SpatialOS documentation)](https://docs.improbable.io/reference/latest/shared/glossary#event) defined on that component.
 
 ## Generation details
-Each struct contains the following fields:
+
+### Snapshot
+
+This struct contains the following fields:
+
+  * the public property `uint ComponentId` to read the component ID of this component as defined in schemalang
+
+Additionally, for each field defined in your schema file, the generated C# struct creates:
+  
+  * a public field corresponding to the field defined in schemalang.
+
+### Component
+
+This struct contains the following fields:
 
   * the public property `uint ComponentId` to read the component ID of this component as defined in schemalang
-  * the public property `BlittableBool DirtyBit` used internally to identify whether a component update needs to be sent to the SpatialOS Runtime
+  * the private fields `byte dirtyBits{i}`, which represents a bitmask used internally to identify whether a component field needs to be replicated to the SpatialOS runtime.
 
 Additionally, for each field defined in your schema file, the generated C# struct creates:
 
   * a private field corresponding to the field defined in schemalang
-  * a public property that can be used for reading and writing the value of this field. Changing the value of this property makes sure that `DirtyBit` is set to `true`.
+  * a public property that can be used for reading and writing the value of this field. Changing the value of this property sets the corresponding `dirtyBit` to true.
```

Comment:
I know the original text says this but, 
...."can be used" is passive voice. Who or what uses it? Does our user use it as part of their development or is it used internally, by the GDK and SpatialOS?
...who changes the value of the property? I guess or user during development. It's clearer to tell them; 
"If you change the value of this property, the corresponding `dirtybit` sets to true.

Reply:
Both the user and the GDK reads and writes the property

Reply:
Can they both change the value of the property too or is it only the lovely user?